src/cascadia/TerminalSettingsEditor/GlobalAppearance.cpp

```diff
@@ -2,19 +2,47 @@
 // Licensed under the MIT license.
 
 #include "pch.h"
+#include "Utils.h"
 #include "GlobalAppearance.h"
 #include "GlobalAppearance.g.cpp"
 #include "MainPage.h"
+#include "EnumEntry.h"
 
 using namespace winrt;
 using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Xaml::Controls;
 using namespace winrt::Microsoft::Terminal::Settings::Model;
+using namespace winrt::Windows::Foundation::Collections;
 
 namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 {
-    GlobalAppearance::GlobalAppearance()
+    GlobalAppearance::GlobalAppearance() :
+        _ElementThemes{ winrt::single_threaded_observable_vector<Editor::EnumEntry>() }
     {
         InitializeComponent();
+
+        auto elementThemeMap = EnumMappings::ElementTheme();
+        for (auto [key, value] : elementThemeMap)
+        {
+            auto enumName = LocalizedNameForEnumName(L"Globals_Theme", value, L"Content");
+            auto entry = winrt::make<EnumEntry>(enumName, winrt::box_value<ElementTheme>(key));
+            _ElementThemes.Append(entry);
+        }
+    }
+
+    IObservableVector<Editor::EnumEntry> GlobalAppearance::ElementThemes()
+    {
+        return _ElementThemes;
+    }
+
+    uint8_t GlobalAppearance::CurrentTheme()
+    {
+        return static_cast<uint8_t>(GlobalSettings().Theme());
+    }
+
+    void GlobalAppearance::CurrentTheme(const uint8_t index)
+    {
+        GlobalSettings().Theme(static_cast<ElementTheme>(index));

```

Comment:
Unfortunately I can't just bind an enum to a XAML property that expects an int (like SelectedIndex) so had to do something like this

Reply:
This also just does not work if the enum isn't consecutive 👀 since I'm binding this property to `SelectedIndex`

Reply:
Hmmmmmm. Okay, how about this? Is it possible for us to bind _selectedITEM_, and then pull the index out of the EnumEntry? We can also search the map to return the EnumEntry for the active theme.

Reply:
oh yeah we could keep a map of enum -> enumentry, and from there we can do `enumMap.Lookup(globalsettings.theme())` to grab the initial `EnumEntry` for the databound `SelectedItem`. Then the TwoWay binding will do all the work as long as we set the xaml->viewmodel binding up so that it updates `GlobalSettings().Theme()`

Reply:
ehh actually scratch that first part, keeping a map just so we can initialize the `SelectedItem` seems overkill

Reply:
ehh screw it i made the map, it avoids having to do if(value), and binding `SelectedItem` avoids the whole index route